oaknational/Oak-Web-Application
# LESQ-1142: Restrict access to hubspot endpoints

## Description

Music year: 2009

- Locks the `/api/hubspot/*` endpoints down to authenticated users
- Users can only request details for their own email address

## How to test

1. While signed in 
2. Go to https://deploy-preview-2928--oak-web-application.netlify.thenational.academy/teachers/programmes/computing-secondary-ks3-l/units/impact-of-technology-collaborating-online-respectfully-35d0/lessons/account-security-68rkee/downloads
3. The page should function as before
4. Requests to `/api/hubspot/contacts` and `/api/hubspot/subscription` in the console should be successful

From the console running:

```typescript
(await fetch("/api/hubspot/contacts")).status
```
```typescript
(await fetch("/api/hubspot/subscription", {
  method: "POST",
  headers: { "Content-Type": "application/json" },
  body: JSON.stringify({ subscriptionName: "School Support" })
})).status
```

Or visiting https://deploy-preview-2928--oak-web-application.netlify.thenational.academy/api/hubspot/contacts

Should return a 20x for each when signed in and a 401 when not 


## Checklist
